org/mozilla/jss/provider/javax/crypto/JSSTokenKeyManager.java

```diff
@@ -118,8 +118,10 @@ public PrivateKey getPrivateKey(String alias) {
 
         try {
             if (jks == null) {
-                org.mozilla.jss.crypto.X509Certificate cert = cm.findCertByNickname(alias);
-                return cm.findPrivKeyByCert(cert);
+                try (PK11Cert cert = (PK11Cert) cm.findCertByNickname(alias)) {
```

Comment:
Does it mean we need to do this in PKI as well? Or will `PK11Cert.close()` get called automatically when it's garbage collected?

Reply:
No, this isn't strictly necessary. It'll get called automatically when it is garbage collected. However, because the scope is small here, it is good to use the try-with-resources pattern to ensure it gets closed and freed quickly. This means the GC pass will just clean up the destroyed Java objects (`PK11Cert`, `CertProxy`, and `TokenProxy` instances), and not need to call into the JNI layer to free NSS stuff.